DOL.WHD.Section14c.Web/src/modules/pages/userLoginPageController.js

```diff
@@ -1,8 +1,38 @@
 'use strict';
 
 module.exports = function(ngModule) {
-  ngModule.controller('userLoginPageController', function() {
+  ngModule.controller('userLoginPageController', function($scope, $location, apiService) {
     'ngInject';
     'use strict';
+
+    var vm = this;
+
+	vm.emailVerificationUrl = $location.absUrl();
+    vm.emailVerificationCode = $location.search().code;
+    vm.emailVerificationUserId = $location.search().userId;
+    vm.isEmailVerificationRequest =
+      vm.emailVerificationCode !== undefined &&
+      vm.emailVerificationCode !== undefined;
```

Comment:
Are both sides of this `&&` supposed to be the same?  Should it be`vm.emailVerificationCode !== undefined && vm.emailVerificationUserId !== undefined`?

I'm also curious if this check should capture `null`, 0-length strings and other falsey conditions.  I.e., should it simply be `vm.emailVerificationCode && vm.emailVerificationUserId`.  A lot of best practices guides say checking explicitly for `undefined` is a code smell, but in practice I know it's sometimes necessary.  What do you think about right here?

Reply:
hmm. I don't remember editing this portion which is odd. I agree with your comments. I think it should be: 
```
 vm.emailVerificationCode || vm.emailVerificationUserId
```
I believe the original expression is just a check to see whether both of them are defined. The expression above implies that if left hand side is undefined or a falsey value, the expression will return as false. If the left hand side is truthy and the right hand side is falsey the expression will return false as well. In addition, if they are both falsey then the epxression will return false. It will only return true if they are both truthy which I think is what we want

Reply:
I agree with these comments. I don't remember editing this portion which is odd.